src/activities/Elsa.Activities.Http/Formatters/XMLContentFormatter.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using Elsa.Activities.Http.Services;
+
+
+namespace Elsa.Activities.Http.Formatters
+{
+    public static class XMLContentFormatterCache
+    {
+        public static List<Type> TypeCache { get; set; }
+    }
+
+
+    public class XMLContentFormatter : IContentFormatter
+    {
+        public int Priority => 1;
+        public IEnumerable<string> SupportedContentTypes => new[] { "application/xml", "text/xml" };
+
+        public Task<object> ParseAsync(byte[] content, string contentType)
+        {
+
+            GetTypeCache();
+
+            string rootNodeTypeName = GetRootTypeName(ref content);
+
+            if (rootNodeTypeName != "")
+            {
+                Type rootType = GetRootType(rootNodeTypeName);
+                XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+                var model = xmlSerializer.Deserialize(new MemoryStream(content));
+                return Task.FromResult<object>(model);
+            }
+
+            // this is not clear - what will be returned if theres no deserialization?
+            return null;
+        }
+
+
+        public Task<byte[]> ParseAsync(object content, string contentType)
+        {
+            Type rootType = content.GetType();
+            XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+            using (MemoryStream stream = new MemoryStream())
+            {
+                xmlSerializer.Serialize(stream, content);
+                return Task.FromResult<byte[]>(stream.ToArray());
+            }
+        }
+
+        private static Type GetRootType(string rootNodeName)
+        {
+            return XMLContentFormatterCache.TypeCache.Where(t => t.Name == rootNodeName).FirstOrDefault();
+        }
+
+        private static void GetTypeCache()
+        {
+            if (XMLContentFormatterCache.TypeCache == null)
+            {
+                string baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+                foreach (string fileName in Directory.EnumerateFiles(baseDirectory, "*.dll", SearchOption.AllDirectories))
+                {
+                    AssemblyLoadContext.Default.LoadFromAssemblyPath(Path.Combine(baseDirectory, fileName));
```

Comment:
I don't like this at all. What's the reason we need to load all assemblies like this?

Reply:
See other comments on this.